### PR TITLE
Volume support

### PR DESCRIPTION
**Pull request by uzaxirr**
### Description of your changes

Fixes #

I have:

- [x] Run `make reviewable test` to ensure this PR is ready for review.
- [x] Read and followed Crossplane's [contribution process].

### How has this code been tested

[contribution process]: https://git.io/fj2m9

### CODE REVIEW

File: apis/civo/volume/v1alpha1/types.go [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+	// +required
+	// +immutable
+	// +kubebuilder:validation:Required
+	Name string `json:"name"`
+
+	// Size for the volume a minimum of 1 and a maximum of your available disk space from your quota specifies the size of the volume in gigabytes
+	// +required
+	// +immutable
+	// +kubebuilder:validation:Required
+	Size string `json:"Size"`
```
Comment (RealHarshThakur) [inline apis/civo/volume/v1alpha1/types.go:49]: size should be lowercase

File: apis/civo/volume/v1alpha1/types.go [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	xpv1 "github.com/crossplane/crossplane-runtime/apis/common/v1"
+)
+
+// CivoVolumeObservation are the observable fields of a CivoVolume.
+type CivoVolumeObservation struct {
+	ID         string `json:"id"`
+	Name       string `json:"name"`
+	ClusterID  string `json:"cluster_id"`
```
Comment (RealHarshThakur) [inline apis/civo/volume/v1alpha1/types.go:26]: remove duplicated fields from spec

File: apis/civo/volume/v1alpha1/types.go [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+	NetworkID  string `json:"network_id"`
+	InstanceID string `json:"instance_id"`
+	Size       string `json:"size"`
+	Status     string `json:"status"`
+	// Details regarding current state of the bucket.
+	Conditions []metav1.Condition `json:"conditions"`
```
Comment (RealHarshThakur) [inline apis/civo/volume/v1alpha1/types.go:32]: we can re-use crospslane conditions

File: apis/civo/volume/v1alpha1/types.go [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+	// +kubebuilder:validation:Required
+	Size string `json:"Size"`
+
+	// NetworkID for the network in which you wish to create the volume.
+	// +required
+	// +immutable
+	// +kubebuilder:validation:Required
+	NetworkID string `json:"network_id"`
+
+	// ProviderReference holds configs (region, API key etc.) for the crossplane provider that is being used.
```
Comment (RealHarshThakur) [inline apis/civo/volume/v1alpha1/types.go:57]: Missing instance id in spec

Comment (uzaxirr) [reply apis/civo/volume/v1alpha1/types.go:67]: https://github.com/crossplane-contrib/provider-civo/pull/58/commits/2c14722448672bb07735b95680c3f2e234bc3ec0

File: apis/civo/volume/v1alpha1/types.go (hunk 2 above)
Comment (uzaxirr) [reply apis/civo/volume/v1alpha1/types.go]: https://github.com/crossplane-contrib/provider-civo/pull/58/commits/38520cf1365b50f297c110d090a14b0ae78b3521

File: internal/controller/civovolume/civovolume.go [hunk 5]
```diff
@@ -0,0 +1,210 @@
...
+	"github.com/crossplane-contrib/provider-civo/apis/civo/volume/v1alpha1"
+	"github.com/crossplane/crossplane-runtime/pkg/event"
+	"github.com/crossplane/crossplane-runtime/pkg/logging"
+	"github.com/crossplane/crossplane-runtime/pkg/reconciler/providerconfig"
+	"github.com/crossplane/crossplane-runtime/pkg/resource"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const (
+	errManagedUpdateFailed          = "cannot update volume custom resource"
```
Comment (RealHarshThakur) [inline internal/controller/civovolume/civovolume.go:39]: error assertion should be done with civogo

**Review by vishalanarase — APPROVED** [review]

**Pull request merged**